### PR TITLE
Sc 43289/globe life unable to pull export

### PR DESCRIPTION
**Pull request by johnrb2**
## Description of the change

> Update to allow `life` as top-level domain; bump mocha to v10.4.0

Up on Dev if there is anyone with a `.life` email address who can test.

Verified that `life` is allowed locally with linked repo in leadconduit-api; added unit test there.

Screenshot 2024-04-04 at 3 15 35 PM

## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Configuration change
- [ ] Technical Debt
- [ ] Documentation

## Related tickets

> https://app.shortcut.com/active-prospect/story/43289/globe-life-unable-to-pull-export

### Development and Testing

- [x]  Lint rules pass locally.
- [x]  The code changed/added as part of this pull request has been covered with tests, or this PR does not alter production code.
- [x]  All tests related to the changed code pass in development, or tests are not applicable.

### Code Review

- [x]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached.
- [x]  At least two engineers have been added as "Reviewers" on the pull request.
- [x]  Changes have been reviewed by at least two other engineers who did not write the code.
- [x]  This branch has been rebased off master to be current.

### Tracking 
- [x]  Issue from Shortcut/Jira has a link to this pull request.
- [x]  This PR has a link to the issue in Shortcut.

### QA
- [x]  This branch has been deployed to staging and tested.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,4 @@
 node_modules
-benchmark/node_modules
```
Comment (johnrb2) [inline .gitignore:2]: the above `node_modules` will ignore `node_modules` recursively throughout the repo

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,4 @@
...
 
 .DS_Store
+.idea
```
Comment (johnrb2) [inline .gitignore:4]: added for it to ignore WebStorm settings

File: package.json [hunk 3]
```diff
@@ -16,7 +16,7 @@
   },
   "devDependencies": {
     "chai": "^1.9.2",
-    "mocha": "^2.0.1"
+    "mocha": "^10.4.0"
```
Comment (johnrb2) [inline package.json:19]: clears `npm` vulnerabilities and tests still pass.

**Review by henriquelakiap — APPROVED** [review]
Nice catches 🥇

File: src/themis.js [hunk 4]
```diff
@@ -481,7 +481,7 @@ var Utils = {
             return true;
         }
         // http://emailregex.com/
-        return /^[-a-z0-9~!$%^&*_=+}{\'?]+(\.[-a-z0-9~!$%^&*_=+}{\'?]+)*@([a-z0-9_][-a-z0-9_]*(\.[-a-z0-9_]+)*\.(aero|arpa|biz|com|coop|edu|gov|info|int|mil|museum|name|net|org|pro|travel|mobi|[a-z][a-z])|([0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}))(:[0-9]{1,5})?$/i.test(email);
+        return /^[-a-z0-9~!$%^&*_=+}{\'?]+(\.[-a-z0-9~!$%^&*_=+}{\'?]+)*@([a-z0-9_][-a-z0-9_]*(\.[-a-z0-9_]+)*\.(aero|arpa|biz|com|coop|edu|gov|info|int|life|mil|museum|name|net|org|pro|travel|mobi|[a-z][a-z])|([0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}))(:[0-9]{1,5})?$/i.test(email);
```
Comment (johnrb2) [inline src/themis.js:484]: The primary fix of this PR

**Review by jgrayson676 — APPROVED** [review]

File: src/themis.js (hunk 4 above)
Comment (ericyd) [reply src/themis.js:484]: This fix is definitely fine, but there are hundreds of top level domains nowadays. I'm not convinced we'll be able to accurately hardcode a valid list of TLDs, I would actually lean towards removing this and adding a more generic TLD check such as `[a-z]{2,10}` (not precisely sure what it should be, just as an example)

Full list of TLDs approved by ICANN: https://data.iana.org/TLD/tlds-alpha-by-domain.txt

File: CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+- bump mocha to ^10.4.0; update npm test script accordingly
+
+## [1.0.1] - 2024-03-11
+## Bump
+- uglify-js to ^2.6
+
+## [1.0.0] - 2024-03-11
+## Fork
+- initial fork of themis@v1.1.6
+- no updates in 9 years from publisher
```
Comment (ericyd) [inline CHANGELOG.md:17]: I do appreciate this comment, but I'm not sold on maintaining a changelog. As demonstrated by this PR, they are easy to forget, and hard to ensure they get updated. Moreover, we don't use them consistently (they are common (but not universal) in our libs, but not used at all in our API and client repos 🤷🏻 why the discrepancy?)

Personally I'd lean towards documenting the "why we forked it" in the readme, and skipping the changelog. I think the git history typically has a very similar summary of changes to the changelog

**Review by ericyd — COMMENTED** [review]
LGTM although I do question whether we should be maintaining a hardcoded list of TLDs at all, it feels like this is going to bite us again and I really can't fathom a benefit to checking for a list of TLDs

File: CHANGELOG.md (hunk 5 above)
Comment (johnrb2) [reply CHANGELOG.md:17]: Yeah that's fair. We can remove in another PR if we want to.

**Comment by johnrb2** [comment]
@ericyd I think we may have a ticket to move off of `themis` anyway. In the mean time I will create another ticket for a more robust solution. ~What do you think about using Node's `dns.lookup()` would that be overkill for something like this or we could add another package that does a similar regex type of email validation?~ I decided `dns.lookup()` probably isn't the best for security reasons.

**Pull request merged**